docs/src/content/en/docs/deployment/server-deployment.mdx

```diff
@@ -1,56 +1,87 @@
 
-See the [Cloud Providers](/docs/deployment/cloud-providers/) for more information.
+## Default project structure
 
-### Building
+The [getting started guide](/docs/getting-started/installation) scaffolds a project with sensible defaults to help you begin quickly. By default, the CLI organizes application files under the `src/mastra/` directory, resulting in a structure similar to the following:
 
-Build the application:
+<FileTree>
+  <FileTree.Folder name="src" defaultOpen>
+    <FileTree.Folder name="mastra" defaultOpen>
+      <FileTree.Folder name="public" />
+      <FileTree.Folder name="agents" />
+      <FileTree.Folder name="tools" />
+      <FileTree.Folder name="workflows" />
+      <FileTree.File name="index.ts" />
+    </FileTree.Folder>
+  </FileTree.Folder>
+  <FileTree.File name="package.json" />
+  <FileTree.File name="tsconfig.json" />
+</FileTree>
+
+## Building
+
+The `mastra build` commands starts the build process:
 
 ```bash copy
-# Build from current directory
 mastra build
+```
 
-# Or specify a directory
-mastra build --dir ./my-project
+### Customizing the input directory
+
+If your Mastra files are located elsewhere, use the `--dir` flag to specify the custom location. The `--dir` flag tells Mastra where to find your entry point file (`index.ts` or `index.js`) and related directories.
+
+```bash copy
+mastra build --dir ./my-project/mastra
 ```
 
-The build process:
+## Build process
 
-1. Locates entry file (`src/mastra/index.ts` or `src/mastra/index.js`)
-2. Creates `.mastra` output directory
-3. Bundles code using Rollup with tree shaking and source maps
-4. Generates [Hono](https://hono.dev) HTTP server
+The build process follows these steps:
 
-See [`mastra build`](/reference/cli/build) for all options.
+1. **Locates entry file**: Finds `index.ts` or `index.js` in your specified directory (default: `src/mastra/`).
+2. **Creates build directory**: Generates a `.mastra/` directory containing:
+    - **`.build`**: Contains dependency analysis, bundled dependencies, and build configuration files.
+    - **`output`**: Contains the production-ready application bundle with `index.mjs`, `instrumentation.mjs`, and project-specific files.
+3. **Copies static assets**: Moves the `public/` folder to the output directory for serving static files.
+4. **Bundles code**: Uses Rollup with tree shaking and source maps for optimization.
+5. **Generates server**: Creates a [Hono](https://hono.dev) HTTP server ready for deployment.
 
-### Running the Server
+### Build output structure
+
+After building, Mastra creates a `.mastra/` directory with the following structure:
+
+<FileTree>
+  <FileTree.Folder name=".mastra" defaultOpen>
+    <FileTree.Folder name=".build" />
+    <FileTree.Folder name="output" defaultOpen>
+      <FileTree.File name="index.mjs" />
+      <FileTree.File name="instrumentation.mjs" />
+      <FileTree.Folder name="public" />
```

Comment:
We don't move the folder but copy the contents over to the `output` folder.

So let's say we have `src/mastra/public/foo.md`. It'll be copied over to `.mastra/output/foo.md`